network/manager.go

```diff
@@ -354,11 +403,56 @@ func (nm *networkManager) CreateEndpoint(cli apipaClient, networkID string, epIn
 	return nil
 }
 
+// UpdateEndpointState will make a call to CNS updatEndpointState API in the stateless CNI mode
+// It will add HNSEndpointID or HostVeth name to the endpoint state
+func (nm *networkManager) UpdateEndpointState(ep *endpoint) error {
+	cnilogger.Logger.Info("Calling cns updateEndpoint API with ", zap.String("containerID: ", ep.ContainerID), zap.String("HnsId: ", ep.HnsId), zap.String("HostIfName: ", ep.HostIfName))
+	response, err := nm.CnsClient.UpdateEndpoint(context.TODO(), ep.ContainerID, ep.HnsId, ep.HostIfName)
+	if err != nil {
+		cnilogger.Logger.Error("Update endpoint API returend with error", zap.Error(err))
+		return errors.Wrapf(err, "Get endpoint API returend with error")
+	}
+	cnilogger.Logger.Info("Update endpoint API returend ", zap.String("podname: ", response.ReturnCode.String()))
+	return nil
+}
+
 // DeleteEndpoint deletes an existing container endpoint.
-func (nm *networkManager) DeleteEndpoint(networkID, endpointID string) error {
+func (nm *networkManager) DeleteEndpoint(networkID, endpointID string, epInfo *EndpointInfo) error {
 	nm.Lock()
 	defer nm.Unlock()
 
+	if nm.IsStatelessCNIMode() {
+		nw := &network{
+			Id:           "azure",
```

Comment:
lets not consider `azure` here instead lets write api to get network name.. incase of multitenancy network name has to be derived from vlan id and subnet space

Reply:
I am using the networkId that is passed tot he func for now.